tests/system_tests.py

```diff
@@ -0,0 +1,62 @@
+import os
+import unittest
+import subprocess
+import time
+
+from Excellenteam.final_project_excellenteam.python_client.web_api_client import WebAppClient
+
+# Put here your default path to the .pptx file
+DEFAULT_FILE_PATH = "C:\\Users\\User\\Desktop\\excellenteam\\excellenteam_python\\t.pptx"
+BASE_URL = 'http://localhost:5000'
+
+client = WebAppClient(BASE_URL)
+
+
+class TestSystem(unittest.TestCase):
+    """
+    A test class to perform system tests for the web application and presentation chat helper.
+
+    Note:
+        This test class relies on the `web_api_client.WebAppClient` to interact with the web application API.
+
+    """
+    @classmethod
+    def setUpClass(cls):
+        # Start the Web API and Explainer in separate subprocesses
+        cls.web_api_process = subprocess.Popen(["python", r"C:\Networks\Excellenteam\final_project_excellenteam\web_api\app.py"])
+        cls.explainer_process = subprocess.Popen(["python", r"C:\Networks\Excellenteam\final_project_excellenteam\explainer\presentation_chat_helper.py"])
+
+        # Wait a few seconds for the servers to start up
+        time.sleep(3)
+
+    @classmethod
+    def tearDownClass(cls):
+        # Terminate the subprocesses for the Web API and Explainer
+        cls.web_api_process.terminate()
+        cls.explainer_process.terminate()
+
+    def test_successful_end_to_end(self):
+        # Upload the file
+        uid = client.upload(DEFAULT_FILE_PATH)
+        self.assertIsNotNone(uid, "UID should not be None")
+        time.sleep(20)
```

Comment:
Assign constants to descriptive variables to enhance readability and simplify future changes ^__^ . Directly using them can lead to confusion and maintenance challenges.